spec/rest/client_spec.rb

```diff
@@ -104,7 +108,9 @@ def initialize(client)
         config.http_client = 'someClient'
         config.region = nil
         config.edge = nil
+        config.logger = nil
       end
+      @client.logger = nil
```

Comment:
Why is this needed? Looks odd.

Reply:
Looking at this. My understanding was the @client should get initialized based on the config set before every example(test), but that is not happening. I am seeing the remnants of the older context/s in the current context/s.

describe Twilio::REST::Client do
  context 'configuration' do
    before do
      Twilio.configure do |config|
        config.region = 'someRegion'
        config.edge = 'someEdge'
        config.logger = 'someLogger'
      end
     it "example 1" do
          something
    end
  end

   context 'validation' do
    before do
      Twilio.configure do |config|
        config.region = nil
        config.edge = nil
        config.logger = nil
      end
    end
    it "example 2" do
       **The client.logger is someLogger and not nil here**
    end
end